app/controllers/proposals_controller.rb

```diff
@@ -148,4 +149,28 @@ def require_waitlisted_or_accepted_state
       redirect_to event_url(@event.slug)
     end
   end
+
+  def require_complete_profile
+    return if current_user.complete? && current_user.unconfirmed_email.blank?
+
+    @incomplete_profile_msg = construct_incomplete_profile_msg.html_safe
```

Comment:
I noticed this is only being used by one action.  This controller already has a lot of before_actions and it might be nicer to have this run inline vs as a before_action.  If we were to do that, we could eliminate using an ivar.  I prefer to avoid ivars in controllers as much as possible.